pkg/image/vulnerability_scan.go

```diff
@@ -0,0 +1,137 @@
+
+	if settings.Ignore != nil {
+		if settings.Ignore.Severity != nil {
+			severity := getSeverityStringForTrivyScan(*settings.Ignore.Severity)
+			trivyArgs = append(trivyArgs, "--severity", severity)
+		}
+		if settings.Ignore.Unfixed != nil {
+			trivyArgs = append(trivyArgs, "--ignore-unfixed", strconv.FormatBool(*settings.Ignore.Unfixed))
+		}
+	}
+
+	cmd := exec.CommandContext(ctx, "trivy", trivyArgs...)
+
+	// Print the command to be executed
+	log.Println(cmd.String())
+
+	cmd.Stdin = nil
+
+	result, err := cmd.CombinedOutput()
+
+	if err != nil {
+		log.Printf("failed to run trivy:\n%s", string(result))
+		return nil, fmt.Errorf("failed to run trivy: %w", err)
+	}
+
+	var trivyResult TrivyResult
+	if err := json.Unmarshal(result, &trivyResult); err != nil {
+		return nil, err
+	}
+
+	// Get the vulnerabilites from the trivy scan
+	vulnerabilities := parseTrivyResult(trivyResult, settings.Ignore)
+
+	// Sort the vulnerabilities by severity
+	severityOrder := map[string]int{"CRITICAL": 0, "HIGH": 1, "MEDIUM": 2, "LOW": 3, "UNKNOWN": 4}
+	sort.Slice(vulnerabilities, func(i, j int) bool {
+		return severityOrder[vulnerabilities[i].Severity] < severityOrder[vulnerabilities[j].Severity]
+	})
+
+	vulnerabilityCountLimit := 50
+	vulnerabilityCountLimitStr := os.Getenv(config.VulnerabilityCountLimitEnvVar)
+	if vulnerabilityCountLimitStr != "" {
+		vulnerabilityCountLimit, err = strconv.Atoi(vulnerabilityCountLimitStr)
+		if err != nil {
+			return nil, fmt.Errorf("failed to get vulnerability count Limit: %w", err)
+		}
+	}
+	if len(vulnerabilities) > vulnerabilityCountLimit {
+		vulnerabilities = vulnerabilities[:vulnerabilityCountLimit]
+	}
+
+	return vulnerabilities, nil
+}
+
+func getSeverityStringForTrivyScan(ignoreSeverities string) string {
+	excludeSeverities := strings.Split(ignoreSeverities, ",")
+
+	// Convert all strings to uppercase
+	for i, str := range excludeSeverities {
+		excludeSeverities[i] = strings.ToUpper(str)
+	}
+
+	supportedSeverities := []string{"UNKNOWN", "LOW", "MEDIUM", "HIGH", "CRITICAL"}
+	excludeSet := make(map[string]bool, len(excludeSeverities))
+	for _, v := range excludeSeverities {
+		excludeSet[v] = true
+	}
+
+	var selected []string
+	for _, value := range supportedSeverities {
+		if !excludeSet[value] {
+			selected = append(selected, value)
+		}
+	}
+	return strings.Join(selected, ",")
+}
+
+func parseTrivyResult(trivyResult TrivyResult, ignoreOptions *buildapi.VulnerabilityIgnoreOptions) []buildapi.Vulnerability {
+	var totalVulnerabilities []TrivyVulnerability
+	for _, result := range trivyResult.Results {
+		totalVulnerabilities = append(totalVulnerabilities, result.Vulnerabilities...)
+	}
+
+	var vulnerabilities []buildapi.Vulnerability
+
+	//create a map for ignored vulnerabilities
+	ignoreMap := make(map[string]bool)
+	if ignoreOptions != nil {
+		for _, vuln := range ignoreOptions.Issues {
+			ignoreMap[vuln] = true
+		}
+	}
+	for _, vuln := range totalVulnerabilities {
```

Comment:
should this loop be inside the validation against not nil for ignoreOptions?

Reply:
No, here I am adding all the vulnerabilities except the ignored ones. The above nil check is just to mark the ignored vulnerabilities.